shared/events.js

```diff
@@ -402,9 +364,22 @@ export class PostMessage extends HashableAction {
   }
 }
 
-export class AuthorizeSender extends HashableAction {
+export class HashableMailboxAuthorizeSender extends HashableAction {
   static authorization = CanModifyAuths
-  static fields = AuthorizeSender.Fields([
+  static fields = HashableMailboxAuthorizeSender.Fields([
     ['sender', 'string']
   ])
 }
+
+// Function for converting from Frontend Contracts
+// TODO Write in a generic way without switch statement
```

Comment:
Please use convention of writing TODOs with a colon, as it helps with editor support. e.g.: `// TODO: Write..`